File: front/apps/web-ele/src/api/core/material_allocation.ts
import { requestClient } from '../request';

export interface MaterialAllocation {
  id: number;
  allocationCode: string;
  orderCode: string;
  materialCode: string;
  allocationQty: number;
  actualQty: number;
  allocationStatus: string;
  allocatedBy: number;
  allocatedAt: string;
  createdBy: number;
  createdAt: string;
  updatedAt: string;
}

export interface MaterialAllocationListResponse {
  total: number;
  list: MaterialAllocation[];
}

export async function createMaterialAllocation(data: {
  allocationCode: string;
  orderCode: string;
  materialCode: string;
  allocationQty: number;
  actualQty?: number;
  allocationStatus: string;
  allocatedBy: number;
  allocatedAt?: string;
  createdBy: number;
}) {
  const response = await requestClient.post<MaterialAllocation>('/material_allocations', data);
  return response;
}

export async function getMaterialAllocations(params?: { limit?: number; offset?: number }) {
  const response = await requestClient.get<MaterialAllocationListResponse>('/material_allocations', {
    params: {
      limit: params?.limit || 100,
      offset: params?.offset || 0,
    },
  });
  return response;
}

export async function getMaterialAllocation(id: number) {
  const response = await requestClient.get<MaterialAllocation>(`/material_allocations/${id}`);
  return response;
}

export async function updateMaterialAllocation(
  id: number,
  data: {
    allocationCode?: string;
    orderCode?: string;
    materialCode?: string;
    allocationQty?: number;
    actualQty?: number;
    allocationStatus?: string;
    allocatedBy?: number;
    allocatedAt?: string;
  }
) {
  const response = await requestClient.post<MaterialAllocation>(`/material_allocations/${id}`, data);
  return response;
}

export async function deleteMaterialAllocation(id: number) {
  const response = await requestClient.delete(`/material_allocations/${id}`);
  return response;
}

File: back/internal/material_allocation/domain/material_allocation.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// MaterialAllocation 胚布调配
type MaterialAllocation struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	AllocationCode string `gorm:"size:50;uniqueIndex;not null" json:"allocationCode"`
	OrderCode string `gorm:"size:50;not null;index" json:"orderCode"`
	MaterialCode string `gorm:"size:50;not null;index" json:"materialCode"`
	AllocationQty float64 `gorm:"type:decimal(10,2);not null" json:"allocationQty"`
	ActualQty float64 `gorm:"type:decimal(10,2)" json:"actualQty"`
	AllocationStatus string `gorm:"size:50;not null;default:'pending'" json:"allocationStatus"`
	AllocatedBy uint `gorm:"not null" json:"allocatedBy"`
	AllocatedAt time.Time `json:"allocatedAt"`
	CreatedBy string         `gorm:"size:50;not null" json:"createdBy"`
	CreatedAt time.Time      `gorm:"autoCreateTime" json:"createdAt"`
	UpdatedAt time.Time      `gorm:"autoUpdateTime" json:"updatedAt"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"deletedAt,omitempty"`
}

// TableName 表名
func (MaterialAllocation) TableName() string {
	return "material_allocations"
}
